src/URI.js

```diff
@@ -1553,6 +1553,7 @@
     var path = this.path();
     var absolute = path.substring(0, 1) === '/';
     var segments = path.split(separator);
+    var noLeadTrailSlashRegex = /^\/*|\/*$/g;
```

Comment:
shouldn't that be `/^\/+|\/+$/g;` (`+` (one or more) instead of `*` (zero or more))?